Source/CarthageKit/Xcode.swift

```diff
@@ -1015,6 +1009,18 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
 			|> filter { projects in !projects.isEmpty }
 			|> flatMap(.Merge) { (projects: [(ProjectLocator, [String])]) -> SignalProducer<(String, ProjectLocator), CarthageError> in
 				return SignalProducer(values: projects)
+					|> flatMap(.Concat) { (project: ProjectLocator, schemes: [String]) in
+						// Only look for schemes that actually reside in the project
```

Comment:
We want to find a scheme that lives in a project (here) but build it through the workspace (below).